metadata/metadata.go

```diff
@@ -318,13 +319,9 @@ func (c *Client) retry(ctx context.Context, cfg requestConfig) (string, error) {
 
 // GetKey gets a specific metadata key.
 func (c *Client) GetKey(ctx context.Context, key string, headers map[string]string) (string, error) {
-	// url.JoinPath first exists in go 1.19
-	key = strings.TrimPrefix(key, "/")
-	var reqURL string
-	if strings.HasSuffix(c.metadataURL, "/") {
-		reqURL = c.metadataURL + key
-	} else {
-		reqURL = c.metadataURL + "/" + key
+	reqURL, err := url.JoinPath(c.metadataURL, key)
+	if err != nil {
+		return "", fmt.Errorf("Failed to form metadata url: %+v", err)
```

Comment:
nit: error strings should not be capitalized. Here and below error return messages

Reply:
Done. Thanks.